src/main/java/org/alfresco/rest/api/impl/NodeDefinitionMapperImpl.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * #%L
+ * Alfresco Remote API
+ * %%
+ * Copyright (C) 2005 - 2020 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.rest.api.impl;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.alfresco.error.AlfrescoRuntimeException;
+import org.alfresco.rest.api.NodeDefinitionMapper;
+import org.alfresco.rest.api.model.NodeDefinitionConstraint;
+import org.alfresco.rest.api.model.NodeDefinition;
+import org.alfresco.rest.api.model.NodeDefinitionProperty;
+import org.alfresco.service.cmr.dictionary.ConstraintDefinition;
+import org.alfresco.service.cmr.dictionary.PropertyDefinition;
+import org.alfresco.service.cmr.dictionary.TypeDefinition;
+import org.alfresco.service.cmr.i18n.MessageLookup;
+import org.alfresco.service.namespace.NamespaceService;
+import org.alfresco.service.namespace.QName;
+/**
+ * Maps representations from TypeDefinition to NodeDefinition
+ *
+ * @author gfertuso
+ */
+public class NodeDefinitionMapperImpl implements NodeDefinitionMapper 
+{
+
+    private final List<String> EXCLUDED_NS = Arrays.asList(NamespaceService.SYSTEM_MODEL_1_0_URI);
```

Comment:
Should I exclude  the properties below or only system properties?
            // top-level minimal info
            ContentModel.PROP_NAME,
            ContentModel.PROP_MODIFIER,
            ContentModel.PROP_MODIFIED,
            ContentModel.PROP_CREATOR,
            ContentModel.PROP_CREATED,
            ContentModel.PROP_CONTENT,
            // other - TBC
            ContentModel.PROP_INITIAL_VERSION,
            ContentModel.PROP_AUTO_VERSION_PROPS,
            ContentModel.PROP_AUTO_VERSION);

Reply:
ContentModel.PROP_CONTENT should be ignored as it's a system managed property (despite the namespace). The rest, in my opinion, should be allowed. However, we should ensure the property definitions are correct i.e. isProtected and isMandatory is true for ContentModel.PROP_MODIFIED (see other inline comment about more tests).

Reply:
Obviously the UI should render these properties appropriately to stop them from being updated (the repo will throw an error if an attempt is made) but having these present will allow clients to determine this dynamically rather than hardcoding behaviour of certain properties.